src/KafkaFlow/Consumers/ConsumerWorker.cs

```diff
@@ -47,7 +47,7 @@ public Task StartAsync()
         {
             this.stopCancellationTokenSource = new CancellationTokenSource();
 
-            this.backgroundTask = Task.Factory.StartNew(
```

Comment:
`Task.Factory.StartNew` should be used for long running *synchronous* work, given the delegate passed in is async, it will effectively return near immediately. [See the second lightbulb comment here](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#avoid-using-taskrun-for-long-running-work-that-blocks-the-thread).

Reply:
Nice catch. I think the same could be done in the ConsumerWorker too, what do you think? Can you do it in this PR?